src/main/kotlin/de/itemis/mps/gradle/generate/Plugin.kt

```diff
@@ -6,41 +6,47 @@ import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.Task
 import org.gradle.api.artifacts.Configuration
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ListProperty
 import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.Optional
 import org.gradle.kotlin.dsl.support.zipTo
 import java.io.ByteArrayOutputStream
 import java.io.File
 import java.util.zip.ZipInputStream
+import javax.inject.Inject
 
 
-open class GeneratePluginExtensions: BasePluginExtensions() {
-    var models: List<String> = emptyList()
+open class GeneratePluginExtensions @Inject constructor(of: ObjectFactory) : BasePluginExtensions(of) {
+    @get:Input
+    @get:Optional
+    val models: ListProperty<String> = of.listProperty(String::class.java)
 }
 
 open class GenerateMpsProjectPlugin : Plugin<Project> {
-
     override fun apply(project: Project) {
         project.run {
             val extension = extensions.create("generate", GeneratePluginExtensions::class.java)
 
             afterEvaluate {
-                val mpsLocation = extension.mpsLocation ?: File(project.buildDir, "mps")
+                val mpsLocation = extension.mpsLocation.map { it.asFile }.getOrElse(File(project.buildDir, "mps"))
```

Comment:
I think this should be solved using a convention for the mpsLocation property.

Reply:
The 'getOrElse(File(project.buildDir, "mps"))'-part should be exactly for this. Isn't that the same behavior as in the original code?

Reply:
I refer to https://docs.gradle.org/current/userguide/lazy_configuration.html#applying_conventions -- this is the standard mechanism to apply default values to lazy properties. But we can leave this for later.

Reply:
Where possible I used the 'convention'. But sometimes I need access to 'project', then it was not possible.